src/databases/hashmap_db.rs

```diff
@@ -12,20 +22,18 @@ use crate::{
 pub struct HashMapDbError {}
 
 impl Display for HashMapDbError {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "")
     }
 }
 
-impl Error for HashMapDbError {}
-
 impl From<HashMapDbError> for BonsaiStorageError {
     fn from(err: HashMapDbError) -> Self {
         Self::Database(err.to_string())
     }
 }
 
-#[derive(Clone)]
+#[derive(Clone, Default)]
```

Comment:
Is it really required?

Reply:
is it a bad thing to have it ?

Reply:
Well it's an empty struct so it can be initialized this way `HashMapDbConfig` wich is as good, if not better as `HashMapDbConfig::default`.

But more than anything `HashMapDbConfig` can be removed from the codebase entirely, can't it? It's a config without fields. There is no reason to store that in our `HashMapDb` struct.

Reply:
removed